packages/ckeditor5-list/tests/manual/listmocking.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * @license Copyright (c) 2003-2021, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/* globals console, window, document */
+
+import ClassicEditor from '@ckeditor/ckeditor5-editor-classic/src/classiceditor';
+import Enter from '@ckeditor/ckeditor5-enter/src/enter';
+import Typing from '@ckeditor/ckeditor5-typing/src/typing';
+import Heading from '@ckeditor/ckeditor5-heading/src/heading';
+import Paragraph from '@ckeditor/ckeditor5-paragraph/src/paragraph';
+import Undo from '@ckeditor/ckeditor5-undo/src/undo';
+import Clipboard from '@ckeditor/ckeditor5-clipboard/src/clipboard';
+import Indent from '@ckeditor/ckeditor5-indent/src/indent';
+import {
+	parse as parseModel,
+	setData as setModelData,
+	getData as getModelData
+} from '@ckeditor/ckeditor5-engine/src/dev-utils/model';
+
+import { modelList, stringifyList } from './../documentlist/_utils/utils';
+import DocumentList from '../../src/documentlist';
+
+ClassicEditor
+	.create( document.querySelector( '#editor' ), {
+		plugins: [ Enter, Typing, Heading, Paragraph, Undo, Clipboard, DocumentList, Indent ],
+		toolbar: [ 'heading', '|', 'bulletedList', 'numberedList', 'outdent', 'indent', '|', 'undo', 'redo' ]
+	} )
+	.then( editor => {
+		window.editor = editor;
+
+		const model = '<paragraph listIndent="0" listItemId="000" listType="bulleted">A</paragraph>\n' +
+		'<paragraph listIndent="0" listItemId="000" listType="bulleted">B</paragraph>\n' +
+		'<paragraph listIndent="1" listItemId="002" listType="numbered">C</paragraph>\n' +
+		'<paragraph listIndent="2" listItemId="003" listType="numbered">D</paragraph>\n' +
+		'<paragraph listIndent="0" listItemId="004" listType="bulleted">E</paragraph>\n' +
+		'<paragraph listIndent="0" listItemId="005" listType="bulleted">F</paragraph>';
+
+		document.getElementById( 'data-input' ).value = model;
+		document.getElementById( 'btn-process-input' ).click();
+	} )
+	.catch( err => {
+		console.error( err.stack );
+	} );
+
+const copyOutput = async () => {
+	const output = document.getElementById( 'data-output' ).innerText;
+
+	await window.navigator.clipboard.writeText( output );
+
+	const copyButton = document.getElementById( 'btn-copy-output' );
+	const label = document.createElement( 'span' );
+
+	label.id = 'btn-copy-label';
+	label.innerText = 'Copied!';
+
+	copyButton.appendChild( label );
+
+	window.setTimeout( () => {
+		label.className = 'hide'; }
+	, 0 );
+
+	window.setTimeout( () => {
+		label.remove();
+	}, 1000 );
+};
+
+const getListModelWithNewLines = stringifiedModel => {
+	return stringifiedModel.replace( /<\/paragraph>/g, '</paragraph>\n' );
+};
+
+const setModelDataFromAscii = () => {
+	const asciiList = document.getElementById( 'data-input' ).value;
+	const cleanedAsciiList = asciiList.replace( /[+|'|\t|;|,]/g, '' );
```

Comment:
Let's narrow it down to leading and trailing characters only. Otherwise, things in the editor content may get stripped out.